src/tcp_wrap.cc

```diff
@@ -131,14 +139,28 @@ void TCPWrap::New(const FunctionCallbackInfo<Value>& args) {
   // normal function.
   CHECK(args.IsConstructCall());
   Environment* env = Environment::GetCurrent(args);
-  new TCPWrap(env, args.This());
+
+  int type_value = args[0]->IntegerValue(env->context()).ToChecked();
```

Comment:
If you want to avoid confusion about the argument type: 

```c++
CHECK(args[0]->IsInt32());
int type_value = args[0].As<Int32>()->Value();
```

:)

(I prefer that style in general, btw.)

Reply:
Good suggestion :)